packages/aragon-wrapper/src/index.js

```diff
@@ -222,39 +222,61 @@ export default class Aragon {
    *         The address of the proxy to get metadata from
    * @return {Promise<Object>}
    */
-  async getAppProxyValues (proxyAddress) {
+  async getProxyValues (proxyAddress) {
     // This function caches information about the AppProxy, as it is called for
     // all the apps everytime a permission changes and this data won't change
     // once it's fetched
-    const cachedValue = appProxyValuesCache[proxyAddress]
+    const cachedValue = proxyValuesCache[proxyAddress]
     if (cachedValue && cachedValue.kernelAddress && cachedValue.appId && cachedValue.codeAddress) {
       return cachedValue
     }
 
-    const appProxy = makeProxy(proxyAddress, 'AppProxy', this.web3, this.kernelProxy.initializationBlock)
-
-    const appProxyValues = await Promise.all([
-      appProxy.call('kernel').catch(() => null),
-      appProxy.call('appId').catch(() => null),
-      appProxy
-        .call('implementation')
-        .catch(() => appProxy
-          // Fallback to old non-ERC897 proxy implementation
-          .call('getCode')
-          .catch(() => null)
-        ),
-      appProxy.call('isForwarder').catch(() => false)
-    ]).then((values) => ({
-      proxyAddress,
-      kernelAddress: values[0],
-      appId: values[1],
-      codeAddress: values[2],
-      isForwarder: values[3]
-    }))
-
-    appProxyValuesCache[proxyAddress] = appProxyValues
+    let proxyValues
+
+    if (addressesEqual(proxyAddress, this.kernelProxy.address)) {
+      const kernelProxy = makeProxy(proxyAddress, 'AppProxy', this.web3, this.kernelProxy.initializationBlock)
+
+      proxyValues = await Promise.all([
+        // Use Kernel ABI
+        this.kernelProxy.call('KERNEL_APP_ID').catch(() => null),
+        // Use ERC897 proxy ABI
+        kernelProxy
+          .call('implementation')
+          .catch(() => kernelProxy
+            // Fallback to old non-ERC897 proxy implementation
+            .call('getCode')
+            .catch(() => null)
```

Comment:
Note that, this bit doesn't work at all for 0.5 organizations built on aragonOS 3.1.0, as `KernelProxy` didn't expose a standard way of getting its backing `Kernel` contract.

It now implements ERC897, so `implementation()` will fetch its `codeAddress`.